src/sentry/workflow_engine/handlers/detector/stateful.py

```diff
@@ -248,44 +261,49 @@ def get_state_data(
     DetectorHandler[DataPacketType, DataPacketEvaluationType],
     abc.ABC,
 ):
-    def __init__(self, detector: Detector):
+    """
+    Stateful Detectors are provided as a base class for new detectors that need to track state.
+    """
+
+    thresholds: DetectorThresholds = {}
+
+    def __init__(self, detector: Detector, thresholds: DetectorThresholds | None = None):
         super().__init__(detector)
-        # if detector.config.get("thresholds"):
-        #     raise Exception(
-        #         "Stateful detectors are required to have `thresholds` set in the config"
-        #     )
 
-        counter_names = self.counter_names or []
-        self.state_manager = DetectorStateManager(detector, counter_names)
+        # Default to 1 for all the possible levels on a given detector
+        default_thresholds = {level: 1 for level in self._get_configured_detector_levels()}
 
-    @property
-    @abc.abstractmethod
-    def counter_names(self) -> list[str]:
-        """
-        The names of the counters that this detector tracks. This is used to build the redis keys for
-        storing counter values.
-        """
-        pass
+        self._thresholds: DetectorThresholds = {
+            DetectorPriorityLevel.OK: 1,  # Make sure the OK level is always set
+            **default_thresholds,
+            **(self.thresholds or {}),  # Allow each handler to override
+            **(thresholds or {}),  # Allow each instance to override
+        }
 
-    def build_issue_fingerprint(self, group_key: DetectorGroupKey = None) -> list[str]:
-        return []
+        self.state_manager = DetectorStateManager(detector, list(self._thresholds.keys()))
 
-    def create_resolve_message(self) -> StatusChangeMessage:
-        """
-        Create a resolve message for the detectors issue. This is overridable in the subclass, but
-        will work for the majority of cases.
-        """
-        return StatusChangeMessage(
-            fingerprint=[*self.build_issue_fingerprint(), self.state_manager.build_key()],
-            project_id=self.detector.project_id,
-            new_status=GroupStatus.RESOLVED,
-            new_substatus=None,
-        )
+    def _get_configured_detector_levels(self) -> list[DetectorPriorityLevel]:
+        # TODO - Is this something that should be provided by the detector itself rather
+        # than having to query the db for each level?
+        priority_levels: list[DetectorPriorityLevel] = [level for level in DetectorPriorityLevel]
+
+        if self.detector.workflow_condition_group is None:
+            # TODO - Should this default to _all_ levels or no levels?
+            return priority_levels
+
+        condition_result_levels = self.detector.workflow_condition_group.conditions.filter(
+            condition_result__in=priority_levels
+        ).values_list("condition_result", flat=True)
+
+        return list(DetectorPriorityLevel(level) for level in condition_result_levels)
```

Comment:
@wedamija this is how i was thinking we could filter to just the levels configure -- any ways we can easily cache this data on the model or should i make a field in the model that looks up all these values and caches it there? 🤔